cla_backend/apps/reports/views.py

```diff
@@ -184,19 +184,29 @@ def all_knowledgebase_articles(request):
 
 @staff_member_required
 def download_file(request, file_name="", *args, **kwargs):
-    conn = get_s3_connection()
-    bucket = conn.lookup(settings.AWS_REPORTS_STORAGE_BUCKET_NAME)
-    k = bucket.get_key(settings.EXPORT_DIR + file_name)
+    # check if there is a connection to aws, otherwise download from local TEMP_DIR
+    if settings.AWS_REPORTS_STORAGE_BUCKET_NAME:
+        conn = get_s3_connection()
+        bucket = conn.lookup(settings.AWS_REPORTS_STORAGE_BUCKET_NAME)
+        k = bucket.get_key(settings.EXPORT_DIR + file_name)
 
-    if k is None:
-        raise Http404("Export does not exist")
+        if k is None:
+            raise Http404("Export does not exist")
+
+        k.open_read()
+        headers = dict(k.resp.getheaders())
+        response = HttpResponse(k)
+
+        for key, val in headers.items():
+            response[key] = val
+    else:
 
-    k.open_read()
-    headers = dict(k.resp.getheaders())
-    response = HttpResponse(k)
+        def _filepath(filename):
+            return os.path.join(settings.TEMP_DIR, filename)
```

Comment:
This might look a bit dangerous, as we are returning them any file they want, nothing stopping them traversing backwards to a specific file(i.e ../../../../../)